modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java

```diff
@@ -0,0 +1,57 @@
+package org.testcontainers.redpanda;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.ComparableVersion;
+import org.testcontainers.utility.DockerImageName;
+
+import java.nio.charset.StandardCharsets;
+
+public class RedpandaContainer extends GenericContainer<RedpandaContainer> {
+
+    private static final DockerImageName REDPANDA_IMAGE = DockerImageName.parse("vectorized/redpanda");
+
+    private static final int REDPANDA_PORT = 9092;
+
+    private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
+
+    public RedpandaContainer(String image) {
+        this(DockerImageName.parse(image));
+    }
+
+    public RedpandaContainer(DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(REDPANDA_IMAGE);
+
+        boolean isLessThan22_2_1 = new ComparableVersion(imageName.getVersionPart()).isLessThan("v22.2.1");
+        if (isLessThan22_2_1) {
+            throw new IllegalArgumentException("Redpanda version must be >= v22.2.1");
+        }
+
+        withExposedPorts(REDPANDA_PORT);
+        withCreateContainerCmdModifier(cmd -> {
+            cmd.withEntrypoint("sh");
+        });
+        waitingFor(Wait.forLogMessage(".*Started Kafka API server.*", 1));
+        withCommand("-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
+    }
+
+    @Override
+    protected void containerIsStarting(InspectContainerResponse containerInfo) {
+        super.containerIsStarting(containerInfo);
+
+        String command = "#!/bin/bash\n";
+
+        command += "/usr/bin/rpk redpanda start --mode dev-container ";
+        command += "--kafka-addr PLAINTEXT://0.0.0.0:29092,OUTSIDE://0.0.0.0:9092 ";
+        command += "--advertise-kafka-addr PLAINTEXT://kafka:29092,OUTSIDE://" + getHost() + ":" + getMappedPort(9092);
+
+        copyFileToContainer(Transferable.of(command.getBytes(StandardCharsets.UTF_8), 0777), STARTER_SCRIPT);
```

Comment:
```suggestion
        copyFileToContainer(Transferable.of(command, 0777), STARTER_SCRIPT);
```

Reply:
there is no `Transferable.of(String, int)`

Reply:
Ahhh.... Let's add one then 😀

Reply:
waiting for https://github.com/testcontainers/testcontainers-java/pull/5741